### PR TITLE
Analysis: Additions on Feature Importance, Predictions, and Prediction Related Findings / Next Steps

### PR DESCRIPTION
**Pull request by JennyonOort**
Second part of analysis added, specifically:
- feature importance exploration thorugh coefficients of the best performing logistic model
- predictions with best model on test data
- report scoring of test data prediction using accuracy and f2 score
- predictions results exploration using confusion matrix, PR curve, and ROC curve
- additional directions to explore as final wrap up of code narrative

Interesting but unexpected finding: BloodPressure (-0.21) actually exhibits negative influences to predicting diabetes, even though effect not as pronounced. This is actually quite counter intuitive to my experience and basic understanding. 

Maybe this will change as we update the model to incorporate review comments. Otherwise, it could also be issues with our coding so we should be more careful to review the code.

### CODE REVIEW

File: analysis/diabetes_analysis.ipynb [hunk 1]
```diff
@@ -0,0 +1,1602 @@
...
+    "    train_test_split,\n",
+    ")\n",
+    "from sklearn.preprocessing import StandardScaler\n",
+    "from sklearn.pipeline import Pipeline, make_pipeline\n",
+    "from sklearn.compose import make_column_transformer\n",
+    "from sklearn.linear_model import LogisticRegression\n",
+    "from sklearn.metrics import (\n",
+    "    fbeta_score, \n",
+    "    confusion_matrix,\n",
+    "    ConfusionMatrixDisplay,\n",
```
Comment (kuo4230) [inline analysis/diabetes_analysis.ipynb:26]: Maybe add a caption under each display figure (i.e. Fig 1. results from the confusion matrix shows... ) to help the reader understand what's happening in the displayed figure.

Comment (kuo4230) [inline analysis/diabetes_analysis.ipynb:26]: Maybe add a caption under each display figure (i.e. Fig 1. results from the confusion matrix shows... ) to help the reader understand what's happening in the displayed figure.

File: analysis/diabetes_analysis.ipynb [hunk 2]
```diff
@@ -0,0 +1,1602 @@
...
+    "    PrecisionRecallDisplay,\n",
```
Comment (kuo4230) [inline analysis/diabetes_analysis.ipynb:27]: Maybe add a caption under each display figure (i.e. Fig 1. results from the confusion matrix shows... ) to help the reader understand what's happening in the displayed figure.

File: analysis/diabetes_analysis.ipynb [hunk 3]
```diff
@@ -0,0 +1,1602 @@
...
+    "    RocCurveDisplay\n",
```
Comment (kuo4230) [inline analysis/diabetes_analysis.ipynb:28]: Maybe add a caption under each display figure (i.e. Fig 1. results from the confusion matrix shows... ) to help the reader understand what's happening in the displayed figure.

File: analysis/diabetes_analysis.ipynb
Comment (kuo4230) [inline analysis/diabetes_analysis.ipynb:1]: Maybe add a caption under each display figures (i.e. Fig 1. results from the confusion matrix shows..., Fig 2, the precision-recall curve shows..., Fig 3, the ROC curve is showing ...) to help the reader understand what's happening in the displayed figures.

**Review by kuo4230 — CHANGES_REQUESTED** [review]
Hi Jenny, 

The coding part looks good to me and narrative is fair. I think adding some figure captions will help make the charts/diagrams more readable for our audience who may not have the necessary knowledge understanding what is happening. I see that there are some more editing on the final narrative part. is it something you are still working on?

File: analysis/diabetes_analysis.ipynb
Comment (JennyonOort) [reply analysis/diabetes_analysis.ipynb:1]: Figure numbering has been added in figure and text for clearer guidance to audience.

Do note that as we discussed and per msg from Tiffany on Slack, I have removed confusion matrix, PR curve, ROC curve as they are 573 materials and have instead presented other visuals to convey similar findings. 

In addition, f2 scores have been taken out as part of analysis as that is also part of 573 knowledge.

**Review by kuo4230 — APPROVED** [review]
Updates look good! Approve for merge

**Review by kuo4230 — APPROVED** [review]
Approving changes.
